### ISSUE TITLE
False Negative Missing Type Variable for library generic types

### ISSUE THREAD

**Issue by hmc-cs-mdrissi** [microsoft/pyright#3315]
**Describe the bug**
reportMissingTypeArgument rule seems to not warn for missing arguments for some generic types. I've mainly noticed this for library packages I have installed sometimes there's a missing warning. Unsure if there's a more precise rule for what kind of generics trigger this. An example is np.ndarray type is generic, but the code below does not show any warning for lack of argument.

```python
from typing import Generic, TypeVar

import numpy as np

T = TypeVar("T")


class Foo(Generic[T]):
    ...


def func1() -> Foo:  # Good error about missing type argument
    ...


def func2() -> np.ndarray:  # Weird, no error for missing type arguments.
    ...
```

**To Reproduce**
Checked with pyright 1.1.235. My pyrightconfig here focuses specifically on "reportMissingTypeArgument" rule here so I set it to,

```json
{
    "typeCheckingMode": "basic",
    "reportMissingTypeArgument": true
}
```

If you apply strict this does warn with one of the unknown rules (reportUnknownVariableType), but I'd expect MissingTypeArgument to be enough to catch this.

**Expected behavior**
Error to appear when np.ndarray is used as an annotation without type arguments.

**VS Code extension or command-line**
pyright 1.1.235. numpy is 1.21.5 and is py.typed library.

**Comment by erictraut** [comment]
Thanks for the bug report. This affects the case where the expression is a member access expression, the LHS is a module, and the RHS is a generic type. My unit tests missed this case.

This will be fixed in the next release.

**Comment by erictraut** [comment]
This is fixed in pyright 1.1.236, which I just published. It will also be included in the next release of pylance.
